src/Fuse/ShaderNode.cs

```diff
@@ -88,7 +90,8 @@ public List<TResourceType> ResourceForTree<TResourceType>(string theResourceId)
             Trees.ReadOnlyTreeNode.Flatten(this).ForEach(n =>
             {
                 if (!(n is AbstractShaderNode input) || !input.Resources.ContainsKey(theResourceId)) return;
-                input.Resources[theResourceId].ForEach<TResourceType>(i => result.Add(i));
+                Stride.Core.Extensions.EnumerableExtensions.ForEach<TResourceType>(input.Resources[theResourceId], i => result.Add(i));
```

Comment:
was this from stride before? maybe `AddRange` works here?

Reply:
i just changed this line.

hm, i didn't find a (quick) way to change this...
but for me, this part looks like it could be improved anyway.

but i feel that is out of the scope of this PR and should go into it's own discussion

Reply:
agreed!